src/actions/actions.ts

```diff
@@ -2082,7 +2082,7 @@ export class UpperCaseOperator extends BaseOperator {
       await TextEditor.replace(range, text.toUpperCase());
 
       vimState.currentMode = ModeName.Normal;
-      vimState.cursorPosition = start;
+      vimState.cursorPosition = vimState.cursorPositionJustBeforeAnythingHappened[0];
```

Comment:
this entire command is already not multicursor friendly though

start in the case of gUU is start of line and end is end of line

gUU is not supposed to move the cursor so it needs to perform the action and remain in place